addons/controllers/antreaconfig_controller_test.go

```diff
@@ -0,0 +1,148 @@
+
+			key := client.ObjectKey{
+				Namespace: "default",
+				Name:      configCRName,
+			}
+
+			cluster := &clusterapiv1beta1.Cluster{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, key, cluster)
+				return err == nil
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			config := &cniv1alpha1.AntreaConfig{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, key, config)
+				if err != nil {
+					return false
+				}
+
+				// Check owner reference
+				if len(config.OwnerReferences) == 0 {
+					return false
+				}
+				Expect(len(config.Finalizers)).Should(Equal(1))
+				Expect(len(config.OwnerReferences)).Should(Equal(1))
+				Expect(config.OwnerReferences[0].Name).Should(Equal(configCRName))
+
+				// TODO: Possible to add more checks here
+				Expect(config.Spec.Antrea.AntConfig.TrafficEncapMode).Should(Equal("encap"))
+				Expect(config.Spec.Antrea.AntConfig.FeatureGates.AntreaTraceflow).Should(Equal(false))
+				Expect(config.Spec.Antrea.AntConfig.FeatureGates.AntreaPolicy).Should(Equal(true))
+				Expect(config.Spec.Antrea.AntConfig.FeatureGates.FlowExporter).Should(Equal(false))
+
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				cluster := &clusterapiv1beta1.Cluster{}
+				err := k8sClient.Get(ctx, key, cluster)
+				if err != nil {
+					return false
+				}
+
+				serviceCIDR, serviceCIDRv6, err := util.GetServiceCIDRs(cluster)
+				if err != nil {
+					return false
+				}
+
+				infraProvider, err := util.GetInfraProvider(cluster)
+				if err != nil {
+					return false
+				}
+
+				// Check infraProvider values
+				Expect(infraProvider).Should(Equal("docker"))
+
+				// Check ServiceCIDR and ServiceCIDRv6 values
+				Expect(serviceCIDR).Should(Equal("192.168.0.0/16"))
+				Expect(serviceCIDRv6).Should(Equal("fd00:100:96::/48"))
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				secretKey := client.ObjectKey{
+					Namespace: "default",
+					Name:      util.GenerateDataValueSecretNameFromAddonAndClusterNames(configCRName, constants.AntreaAddonName),
+				}
+				secret := &v1.Secret{}
+				err := k8sClient.Get(ctx, secretKey, secret)
+				if err != nil {
+
+					return false
+				}
+				Expect(secret.Type).Should(Equal(v1.SecretTypeOpaque))
+				secretData := string(secret.Data["values.yaml"])
+				Expect(strings.Contains(secretData, "AntreaPolicy: true")).Should(BeTrue())
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				// Check status.secretRef after reconciliation
+				config := &cniv1alpha1.AntreaConfig{}
+				err := k8sClient.Get(ctx, key, config)
+				if err != nil {
+					return false
+				}
+				Expect(config.Status.SecretRef).Should(Equal(util.GenerateDataValueSecretNameFromAddonAndClusterNames(configCRName, constants.AntreaAddonName)))
```

Comment:
Could we also assert contents of the antreaSecret?

Reply:
Added check